### ISSUE TITLE
Functions not supported as values in Squiz ArrayDeclaration sniff

### ISSUE THREAD

**Issue by keyermann** [squizlabs/PHP_CodeSniffer#500]
Hi,

I have a serious problem with Squiz sniffer and fix when I have an array with function.

Exemple : 

```php
<?php
    array(
        'key1' => function($bar) {
            return $bar;
        },
        'key2' => function($foo) {
            return $foo;
        },
    );
```

Result with Squiz : 

```php
<?php

    array(
     'key1' => function($bar) {
            ,
            return $bar;
        },
     'key2' => function($foo) {
            ,
            return $foo;
        },
    );
```

Is there a way to correct it ?

Thank you ! ;)

**Comment by gsherwood** [comment]
That sniff does not support functions being used as array values, so I'll add support for this to the sniff.

**Comment by keyermann** [comment]
Thank you.

I will add an other problem linked : 

with this solution I have a **Squiz.WhiteSpace.ScopeClosingBrace.Indent** error : *Closing brace indented incorrectly; expected 5 spaces, found 8*
```php
<?php

    array(
     'key1' => function($bar) {
            return $bar;
        },
     'key2' => function($foo) {
            return $foo;
        },
    );
```

If I correct it like this : 

```php
<?php

    array(
     'key1' => function($bar) {
            return $bar;
     },
     'key2' => function($foo) {
            return $foo;
     },
    );
```

I have a **Generic.WhiteSpace.ScopeIndent.IncorrectExact** error : *Line indented incorrectly; expected 8 spaces, found 5*

**Comment by gsherwood** [comment]
> I will add an other problem linked :

Please add a new issue for this so it can be tracked. I've fixed the original issue you are having with closures inside array values, but this other issue is unrelated to that.
